File: org.palladiosimulator.blockchainsystems.core/src/main/java/org/palladiosimulator/blockchainsystems/core/transactionpropagation/TransactionPropagationStrategyFactoryImpl.java
package org.palladiosimulator.blockchainsystems.core.transactionpropagation;

import org.palladiosimulator.blockchainsystems.core.system.abstractions.PropagationStrategy;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.PropagationStrategyFactory;
import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction;

public class TransactionPropagationStrategyFactoryImpl implements PropagationStrategyFactory<Transaction> {
    @Override
    public PropagationStrategy<Transaction> createPropagationStrategy() {
        return new TransactionPropagationStrategyImpl();
    }
}

File: org.palladiosimulator.blockchainsystems.core/src/main/kotlin/org/palladiosimulator/blockchainsystems/core/transactionpropagation/TransactionPropagationStrategyFactoryImpl.kt
package org.palladiosimulator.blockchainsystems.core.transactionpropagation

import org.palladiosimulator.blockchainsystems.core.system.abstractions.PropagationStrategy
import org.palladiosimulator.blockchainsystems.core.system.abstractions.PropagationStrategyFactory
import org.palladiosimulator.blockchainsystems.core.system.abstractions.Transaction

class TransactionPropagationStrategyFactoryImpl : PropagationStrategyFactory<Transaction> {
  override fun createPropagationStrategy(): PropagationStrategy<Transaction> {
    return TransactionPropagationStrategyImpl()
  }
}